src/sources/file.rs

```diff
@@ -561,16 +568,26 @@ pub fn file_source(
 
         let mut encoding_decoder = encoding_charset.map(Decoder::new);
 
+        let mut file_id_to_metrics_mapping =
+            HashMap::<FileFingerprint, FileBytesReceivedHandle>::new();
```

Comment:
Take-it-or-leave-it suggestion:
```suggestion
        let mut file_id_to_metrics_mapping: HashMap<FileFingerprint, FileBytesReceivedHandle> =
            HashMap::new();
```
This puts the type name on the variable instead of the function call. It's a minor issue here given how close the two are together, but can be helpful with the likes of `.collect::<Vec<_>>()` at the end of a complex iterator expression.

Reply:
Fixed.